src/msix/PAL/FileSystem/Win32/DirectoryObject.cpp

```diff
@@ -130,12 +131,32 @@ namespace MSIX {
         // When requesting size, length accounts for null char
         result.resize(prefixChars + length, L' ');
 
-        DWORD newlength = GetFullPathNameW(pathWide.c_str(), length, &result[prefixChars], nullptr);
+        DWORD newLength = GetFullPathNameW(pathWide.c_str(), length, &result[prefixChars], nullptr);
 
-        // On success, length does not account for null char
-        ThrowLastErrorIf(length == 0, "Failed to get necessary char count for GetFullPathNameW");
-        ThrowErrorIf(Error::Unexpected, (length - 1) != newlength, "Result length was unexpected");
-        result.resize(prefixChars + newlength);
+        // On success, newLength does not account for null char
+        ThrowLastErrorIf(newLength == 0, "Failed to get necessary char count for GetFullPathNameW");
+
+        // The normal scenario is that length - 1 == newLength, but for relative paths, GetFullPathName
+        // doesn't return the correct case size and there's no guarantee it will be always bigger than needed. 
+        // If we are in a case that length > newlenght there's no harm, just resize the string. Otherwise, it means 
+        // that the first length was not correct and the path didn't get resolved correctly. Try until previous length
+        // is lower than the next call.
+        if (length < newLength)
+        {
+            DWORD retry = 1;
+            DWORD previousLength = 0;
+
+            do
+            {
+                previousLength = newLength + retry;
+                result.resize(prefixChars + previousLength, L' ');
+                newLength = GetFullPathNameW(pathWide.c_str(), previousLength, &result[prefixChars], nullptr);
+                retry++;
+            } while ((previousLength > newLength) && retry <= 10);

```

Comment:
>previousLength > newLength [](start = 22, length = 26)

Shouldn't this be prev < new ?  As in, "if new is bigger, we need to try again".